### PR TITLE
Advisory note to be displayed when the search functionality defaults to Bing instead of Google

### LINKED ISSUE #37451: Advisory note to be displayed when the search functionality defaults to Bing instead of Google
If you use the search functionality on the website and type in `persistent volumes` then the results will populate with identical entries. The results only differ slightly by the query parameters in the URL.

It appears that links with query parameters like `ref` are being included in search results and leading to duplicate results.

image

### PR DESCRIPTION
**Pull request by coder12git**
fixes #37451

This PR displays an advisory note, when the search functionality defaults to `Bing` instead of `Google`.
Below i am attaching the screenshot of the note which is shown when the search is `Bing`. Please note that in case of `Google` , there will be no note.

Screenshot 2023-12-21 231148

### CODE REVIEW

File: static/js/search.js [hunk 1]
```diff
@@ -45,6 +45,8 @@
     }
 
     window.renderBingSearchResults = () => {
+      let note = 'The search results on this page are currently using Bing. Please note that Bing search may not be as effective as Google search at indexing websites where different query strings might be used, and the results may vary from your usual experience.';
```
Comment (sftim) [inline static/js/search.js:48]: Can we [localize](https://gohugo.io/content-management/multilingual/#translation-of-strings) this string? I'm not sure how - maybe we can embed the string to use as a hidden / metadata element inside the `<head>` of the search page.

**Comment by sftim** [comment]
/area localization

The state-level censorship is associated with specific locales.

**Comment by coder12git** [comment]
@sftim done the changes, it's good for review now ;)

File: data/i18n/en/en.toml [hunk 2]
```diff
@@ -9,6 +9,9 @@ other = "(auto-generated page)"
 [auto_generated_pageinfo]
 other = """<p>This page is automatically generated.</p><p>If you plan to report an issue with this page, mention that the page is auto-generated in your issue description. The fix may need to happen elsewhere in the Kubernetes project.</p>"""
 
+[bing_note]
+other = "The search results on this page are currently using Bing. Please note that Bing search may not be as effective as Google search at indexing websites where different query strings might be used, and the results may vary from your usual experience."
```
Comment (kbhawkey) [inline data/i18n/en/en.toml:13]: I don't think it is a good idea to provide more information about the search tool besides the first sentence,
`The search results on this page are generated by Bing.`

Comment (coder12git) [reply data/i18n/en/en.toml:13]: Sure, if it's not required than i will remove that :)

**Comment by coder12git** [comment]
This PR is good for a review now

**Comment by coder12git** [comment]
@sftim can you please review this, I have done the changes as you suggested.

**Comment by sftim** [comment]
I'm happy for anyone to review this. See https://github.com/kubernetes/website/pull/44463#event-11318363671 for the automatically-assigned reviewer.

Pretty much all the reviewers are volunteers, too. Please bear that in mind.

**Comment by coder12git** [comment]
@natalisucks @onlydole can i get a review plz?

File: data/i18n/en/en.toml (hunk 2 above)
Comment (natalisucks) [inline data/i18n/en/en.toml:13]: nit: Google Search is a product itself, so the term "search" should be capitalized as Google Search.

second, we're making an assumption that people looking at the search results on this page usually use Google, but that isn't always the case, and given the geopolitical aspects, this needs to be a neutral statement.

I would suggest "The search results on this page are currently using Bing. Please note that Bing differs from Google Search at indexing websites where different query strings might be used."

**Comment by natalisucks** [comment]
/hold

Given the issue #44475 under current discussion, we should likely hold this PR. Thanks @coder12git for your work here, and please also be aware that the timeframe of this work and your request for review has occurred over a holiday period, so reviews are going to be slower coming in as people ramp up back into work for 2024.

**Comment by divya-mohan0209** [comment]
@coder12git Would you be able to rebase this PR? Additionally, per the update on the issue, there seems to be some progress on implementing PageFind as an alternative, so we might need to revise that bit as well, once the implementation is done.

**Comment by divya-mohan0209** [comment]
@coder12git : I am closing this PR since we no longer default to Bing search instead of Google per [this issue](https://github.com/kubernetes/website/issues/47137). We're using Pagefind wherever Google programmable search isn't working due to firewall/legal restrictions.
/close

**Pull request closed without merging**